adb_client.c

```diff
@@ -362,16 +363,14 @@ static adb_service_t *adb_service_open(adb_client_t *client, const char *name, a
     }
 #endif
 
-    else if (!strncmp(name, "reboot:", 7)) {
+    if (!strncmp(name, "reboot:", 7)) {
         adb_reboot_impl(&name[7]);
 
         /* One shot service, skip service register */
         return NULL;
     }
 
-#if defined(CONFIG_ADBD_SHELL_SERVICE) || defined(CONFIG_ADBD_LOGCAT_SERVICE)
 service_created:
```

Comment:
Compilation may fail if none of the flags that lead to `goto service_created` are defined:
- CONFIG_ADBD_SHELL_SERVICE
- CONFIG_ADBD_LOGCAT_SERVICE
- CONFIG_ADBD_FILE_SERVICE

What about this?:
```
#if defined(CONFIG_ADBD_SHELL_SERVICE) ||
    defined(CONFIG_ADBD_LOGCAT_SERVICE) ||
    defined(CONFIG_ADBD_FILE_SERVICE)
```

Reply:
If all of these services aren't enabled, which mean the user can just issue "adb reboot".  Is it make sense to support this minimal functionality?

Reply:
You are right, it does not make sense to support `reboot` feature only.

Unfortunately this configuration is supported by the build system and this will generate compile error. For this reason I would keep the `#if defined(...)` statement until a better workaround is implemented to avoid the build error.

Reply:
Ok, I fix in another way, please take a look.